src/lib.rs

```diff
@@ -182,6 +182,30 @@ impl PublicKey {
     pub fn to_bytes(self) -> [u8; PK_SIZE] {
         self.0.to_compressed()
     }
+
+    /// Deserialize a hex-encoded representation of a `PublicKey` to a `PublicKey` instance.
+    pub fn from_hex(hex: &str) -> Result<Self> {
+        let pk_bytes = hex::decode(hex).map_err(|e| {
+            Error::HexConversionFailed(format!(
+                "Failed to decode hex representation: {}",
+                e.to_string()
+            ))
+        })?;
+        let pk_bytes: [u8; PK_SIZE] = pk_bytes.try_into().unwrap_or_else(|v: Vec<u8>| {
+            panic!(
+                "Expected vec of length {} but received vec of length {}",
+                PK_SIZE,
+                v.len()
+            )
```

Comment:
Why panic? can't we return an error?

Reply:
Sorry, I had used this mechanism from Stackoverflow. Will change.